src/eventHandlers/nfts.ts

```diff
@@ -122,7 +122,7 @@ export const nftCollectionCreatedHandler = async (event: SubstrateEvent): Promis
     record.nbNfts = 0
     record.hasReachedLimit = false
     record.isClosed = false
-    record.limit = Number(limit.toString()) || null
+    record.limit = isNaN(Number(limit?.toString())) ? null : Number(limit?.toString())
     record.timestampCreate = commonEventData.timestamp
```

Comment:
isNaN could be replaced by either `Number.isNan()` or `Number.isFinite()` to provide more consistency. 
Here a short article about it : https://bobbyhadz.com/blog/javascript-check-if-value-is-number#check-if-a-value-is-a-number-using-numberisfinite